src/darwin/Framework/CHIP/MTRCertificateInfo.mm

```diff
@@ -0,0 +1,185 @@
+/**
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTRCertificateInfo.h"
+
+#import "MTRConversion.h"
+#import "MTRDefines_Internal.h"
+
+#include <credentials/CHIPCert.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+using namespace chip;
+using namespace chip::Credentials;
+using namespace chip::ASN1;
+
+@interface MTRCertificateInfo ()
+- (nullable instancetype)initWithTLVBytes:(MTRCertificateTLVBytes)bytes NS_DESIGNATED_INITIALIZER;
+@end
+
+@interface MTRDistinguishedNameInfo ()
+- (instancetype)initWithDN:(const ChipDN &)dn;
+@end
+
+MTR_DIRECT_MEMBERS
+@implementation MTRCertificateInfo {
+    NSData * _bytes; // needs to be kept around, _data may contain pointers into the backing buffer
+    struct ChipCertificateData _data;
+}
+
+- (nullable instancetype)initWithTLVBytes:(MTRCertificateTLVBytes)bytes
+{
+    if (self = [super init]) {
+        VerifyOrReturnValue(_bytes = [bytes copy], nil);
+        VerifyOrReturnValue(DecodeChipCert(AsByteSpan(_bytes), _data) == CHIP_NO_ERROR, nil);
+    }
+    return self;
+}
+
+- (MTRDistinguishedNameInfo *)issuer
+{
+    return [[MTRDistinguishedNameInfo alloc] initWithDN:_data.mIssuerDN];
+}
+
+- (MTRDistinguishedNameInfo *)subject
+{
+    return [[MTRDistinguishedNameInfo alloc] initWithDN:_data.mSubjectDN];
+}
+
+- (NSDate *)notBefore
+{
+    return ChipEpochSecondsAsDate(_data.mNotBeforeTime);
+}
+
+- (NSDate *)notAfter
+{
+    // "no expiry" is encoded as kNullCertTime (see ChipEpochToASN1Time)
+    return (_data.mNotAfterTime != kNullCertTime) ? ChipEpochSecondsAsDate(_data.mNotAfterTime) : NSDate.distantFuture;
+}
+
+- (id)copyWithZone:(nullable NSZone *)zone
+{
+    return self;
```

Comment:
We don't need to make an actual copy because we have no mutable state?  Might be worth documenting that.